src/websock/native/integrate_test.go

```diff
@@ -0,0 +1,508 @@
+// +build !without_amqp
+
+package native_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/rinq/httpd/src/websock/native"
+	"github.com/rinq/httpd/src/websock/native/message"
+	"github.com/rinq/rinq-go/src/rinq"
+	"github.com/rinq/rinq-go/src/rinq/options"
+	"github.com/rinq/rinq-go/src/rinqamqp"
+	"github.com/satori/go.uuid"
+	"io"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"time"
+)
+
+// absolute hack to get around nil being a valid value
+var noBody *struct{ Foo string } = nil
```

Comment:
Let's get rid of absolute hacks ;) 

As best I can tell this is only used to construct buffers for sync call error. It's clearer to have specific functions in this case. (as in `blahWithPayload()`, `blahWithoutPayload()`. I know it's "only tests" but it's still code that has to be deciphered by others, potentially years later.

Reply:
In fact, for any "outbound" messages such as those errors, the existing `message.Write()` function does exactly what you want. That would mean `msg` would only be used to construct frames for inbound-only i.e. those for which there is no production "writer", which makes a lot of sense.